pkg/reconciler/reconciler.go

```diff
@@ -466,6 +495,16 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.
 	}
 	u.UpdateStatus(updater.EnsureCondition(conditions.Initialized(corev1.ConditionTrue, "", "")))
 
+	for _, ext := range r.preExtensions {
+		if err := ext(ctx, obj, r.log); err != nil {
+			u.UpdateStatus(
+				updater.EnsureCondition(conditions.Irreconcilable(corev1.ConditionTrue, conditions.ReasonReconcileError, err)),
```

Comment:
If there are reasons as detailed as `ReasonErrorGettingClient` or `ReasonErrorGettingValues` then I think extensions deserve their own `ReasonExtensionError`, or perhaps even, a `ReasonPreExtensionError` and `ReasonPostExtensionError`, rather than piggy-back on `ReasonReconcileError`.

Reply:
I would rather not do so, because I do not like these reasons to begin with. They expose implementation details. I would much prefer to actually let users specify their own reasons to be set in cases of errors. Eventually, I would like to get rid of all Helm-operator specific reasons.

`ReasonReconcileError` seems good enough because the extensions implement reconciliation logic themselves and thus are part of the reconciliation process.